src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts

```diff
@@ -1,33 +1,406 @@
+
+    component.loadFeedbackQuestions();
+    expect(component.questionEditFormModels.length).toBe(2);
+    expect(component.questionEditFormModels[0].feedbackQuestionId).toBe('feedback-question-1');
+    expect(component.questionEditFormModels[1].feedbackQuestionId).toBe('feedback-question-2');
+  });
+
+  it('should get all students of the course', () => {
+    const testStudent1: Student = {
+      email: 'alice@tmms.com',
+      courseId: 'testId',
+      name: 'Alice',
+      teamName: 'Team 1',
+      sectionName: 'Section 1',
+    };
+    const testStudent2: Student = {
+      email: 'bob@tmms.com',
+      courseId: 'testId',
+      name: 'Bob',
+      teamName: 'Team 1',
+      sectionName: 'Section 1',
+    };
+    const students: Students = {
+      students: [testStudent1, testStudent2],
+    };
+    spyOn(studentService, 'getStudentsFromCourse').and.returnValue(of(students));
+
+    component.getAllStudentsOfCourse();
+
+    expect(component.studentsOfCourse.length).toBe(2);
+    expect(component.studentsOfCourse[0].name).toBe('Alice');
+  });
+
+  it('should get all instructors of the course', () => {
+    const testInstructor1: Instructor = {
+      courseId: 'testId',
+      email: 'testB@example.com',
+      name: 'Instructor A',
+      joinState: JoinState.JOINED,
+    };
+    const testInstructor2: Instructor = {
+      courseId: 'testId',
+      email: 'testA@example.com',
+      name: 'Instructor B',
+      joinState: JoinState.JOINED,
+    };
+    const instructors: Instructors = {
+      instructors: [testInstructor1, testInstructor2],
+    };
+    spyOn(instructorService, 'loadInstructors').and.returnValue(of(instructors));
+
+    component.getAllInstructorsCanBePreviewedAs();
+    expect(component.instructorsCanBePreviewedAs.length).toBe(2);
+    expect(component.instructorsCanBePreviewedAs[0].name).toBe('Instructor A');
+  });
+
+  it('should collapse all questions', () => {
+    testQuestionEditFormModel1.isCollapsed = false;
+    testQuestionEditFormModel2.isCollapsed = false;
+    component.isLoadingFeedbackQuestions = false;
+    component.questionEditFormModels = [testQuestionEditFormModel1, testQuestionEditFormModel2];
+    fixture.detectChanges();
+
+    const button: any = fixture.debugElement.nativeElement.querySelector('#btn-collapse-expand');
+    button.click();
+
+    expect(component.questionEditFormModels[0].isCollapsed).toBe(true);
+    expect(component.questionEditFormModels[1].isCollapsed).toBe(true);
+  });
+
+  it('should expand all questions when at least one question is collapsed', () => {
+    testQuestionEditFormModel1.isCollapsed = true;
+    testQuestionEditFormModel2.isCollapsed = false;
+    component.questionEditFormModels = [testQuestionEditFormModel1, testQuestionEditFormModel2];
+    component.isLoadingFeedbackQuestions = false;
+    fixture.detectChanges();
+
+    const button: any = fixture.debugElement.nativeElement.querySelector('#btn-collapse-expand');
+    button.click();
```

Comment:
Would have thought to test `collapseAll()` and `expandAll()` individually, but this also looks good to me!